src/task.c

```diff
@@ -255,7 +255,7 @@ task_create_thread_with_hooks(platform_thread       *thread,
       platform_error_log("Cannot create a new thread as the limit on"
                          " concurrent threads, %d, will be exceeded.\n",
                          MAX_THREADS);
-      return STATUS_LIMIT_EXCEEDED;
+      return STATUS_BUSY;
```

Comment:
Thank you for fixing the inconsistency.

But don't you think LIMIT_EXCEEDED is the most appropriate error code here?

Reply:
Unfortunately, we have this mapping in our code base:

``` 25 #define STATUS_LIMIT_EXCEEDED CONST_STATUS(ENOSPC)`

which ends up spitting out a [misleading] `Could not create a thread: No space left on device` message.

I think we should look into properly remapping `STATUS_LIMIT_EXCEEDED` to a more appropriate errno-code, outside this PR (if you don't mind).